### PR TITLE
networkmanager to ignore the configuration of secondary vnics

### PR DESCRIPTION
**Pull request by guidotijskens**
the change adds the keyfile section to the NetworkManager.conf file, it that exists, and adds one or more unmanaged-devices with the mac addresses, on a single line, when configuring;
when deconfiguring, it removes the unmanaged-devices line from the NetworkManager.conf file;
From the help info, the script should be able to configure or deconfigure only one secondary vnic at a time, but I could not find how to let it work, so this is not implemented (yet);
This is not expected to be a big issue, as it would look weird to use networkmanager for 1 device and not for another on a single environment, it would at least make maintenance more complicated.

### CODE REVIEW

**Comment by guidotijskens** [comment]
added code to add/remove in macaddress at a time to the NetworkManager.conf file; as it is not clear to me how this command can be triggered correctly, or to create a test env to accomplish this, this part is only tested off-line.

File: libexec/secondary_vnic_all_configure.sh [hunk 1]
```diff
@@ -597,6 +599,59 @@ oci_vcn_ip_addr_del_iface() {
...
+    oci_vcn_debug "macaddres: $mac"
+    oci_disable_network_mgr_mac $mac
+}
+
+oci_disable_network_mgr_mac(){
+    #
+    # adds the mac address to the unmanaged-devices list in then NetworkManager.conf file.
+    local -r mac=$1
+    oci_vcn_debug "macaddres: $mac"
+    local -r ifname=$1
```
Comment (wbora) [inline libexec/secondary_vnic_all_configure.sh]: This does not seem right, above the mac is $1, and now it is the ifname?

File: libexec/secondary_vnic_all_configure.sh [hunk 2]
```diff
@@ -597,6 +599,59 @@ oci_vcn_ip_addr_del_iface() {
...
+    if [ -f $NWM_CONF ] && [ -r $NWM_CONF ]; then
+        oci_vcn_debug "$NWM_CONF exists and is readable"
+        keyfilepresent=$(grep -i keyfile $NWM_CONF)
+        if [ -z "${keyfilepresent}" ]; then
+             oci_vcn_debug "[keyfile] tag not present"
+             printf '\n[keyfile]\nunmanaged-devices=mac:%s\n' $mac >> $NWM_CONF
+        else
+            oci_vcn_debug "[keyfile] tag present"
+            unmanageddevicespresent=$(grep -i unmanaged-devices $NWM_CONF)
+            oci_vcn_debug "found $unmanageddevicespresent"
```
Comment (wbora) [inline libexec/secondary_vnic_all_configure.sh]: This debuf statement is out of place, the check for this is below this

File: libexec/secondary_vnic_all_configure.sh [hunk 3]
```diff
@@ -712,10 +768,48 @@ oci_vcn_ip_sec_addr_del() {
...
+    local -r mac=$1
+    oci_vcn_debug "macaddres: $mac"
+    if [ -f $NWM_CONF ] && [ -r $NWM_CONF ]; then
+        oci_vcn_debug "$NWM_CONF exists and is readable"
+        keyfilepresent=$(grep -i keyfile $NWM_CONF)
+        if [ -z "${keyfilepresent}" ]; then
+             oci_vcn_debug "[keyfile] tag not present"
+        else
+             unmanageddevicespresent=$(grep -i unmanaged-devices $NWM_CONF)
+             oci_vcn_debug "found $unmanageddevicespresent"
```
Comment (wbora) [inline libexec/secondary_vnic_all_configure.sh]: There is a check (if) missing here...

**Pull request merged**